server/aspnetcore/TicketHub/Controllers/TicketController.cs

```diff
@@ -0,0 +1,110 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Net.Mime;
+using System.Security.Claims;
+using TicketHub.Database;
+
+namespace TicketHub.Controllers;
+
+[ApiController]
+[Route("/api")]
+[Produces("application/json")]
+[Consumes(MediaTypeNames.Application.Json)]
+public class TicketController : ControllerBase
+{
+    private readonly ILogger<TicketController> _logger;
+    private readonly PostgresContext _dbContext;
+
+    public record TicketFields(string customer, string description);
+    public record ResolveFields(bool resolved);
+
+    private async Task<Customer> addCustomer(Tenant tenant, string name)
+    {
+        Customer c = new Customer();
+        c.Name = name;
+        c.Tenant = tenant.Id;
+        await _dbContext.Customers.AddAsync(c);
+        return c;
+    }
+
+    public TicketController(ILogger<TicketController> logger, PostgresContext dbContext)
+    {
+        _logger = logger;
+        _dbContext = dbContext;
+    }
+
+    // List all tickets.
+    [HttpGet]
+    [Route("tickets")]
+    public async Task<ActionResult<IAsyncEnumerable<Ticket>>> ListTickets()
+    {
+        var tName = User.FindFirstValue("Tenant");
+        if (tName is string)
+        {
+            Tenant tenant = await getTenantByName(tName);
+            List<Ticket> tickets = await _dbContext.Tickets

```

Comment:
I am surprised you do not need an import for `System.Collections.Generic` because you are using `List<>` here. 🤔

Reply:
I suspect it is one or more of: implicit `using`s, framework includes, or else a C# 10+ language feature?

In any case, my IDE tooling suggested I didn't need the explicit `System.Collections.Generic` includes, so I pruned them out without any ill effects. :shrug: